docs/dev/datastream_execution_mode.md

```diff
@@ -237,6 +237,35 @@ next key.
 See [FLIP-140](https://cwiki.apache.org/confluence/x/kDh4CQ) for background
 information on this.
 
+### Order of Processing
+
+The order in which records are processed in operators or user-defined functions (UDFs) can differ between `BATCH` and `STREAMING` execution.
+
+In `STREAMING` mode, user-defined functions should not make any assumptions about incoming records' order.
+Data is processed as soon as it arrives.
+
+In `BATCH` execution mode, there are some operations where Flink guarantees order. 
+The ordering can be a side effect of the particular task scheduling,
+network shuffle, and state backend (see above), or a conscious choice by the system.
+
+There are three general types of input that we can differentiate:
+
+- _broadcast input_: input from a broadcast stream (see also [Broadcast
+  State]({% link dev/stream/state/broadcast_state.md %}))
+- _regular input_: input that isn't any of the above types of input
```

Comment:
regular input should be processed before keyed input, because the (sorted) keyed input is always last.

Reply:
I think I might've not been clear on this one. What I meant is that the line with `regular input` should be below the `keyed input`, because it refers to it. If I understood the sentence correctly it says: `input that isn't broadcast or keyed`, right?

Reply:
Ah damn, you're right... 🙈 fixing it